packages/beacon-node/src/execution/builder/cache.ts

```diff
@@ -0,0 +1,34 @@
+import {BLSPubkey, Epoch, bellatrix} from "@lodestar/types";
+import {toPubkeyHex} from "@lodestar/utils";
+
+const REGISTRATION_PRESERVE_EPOCHS = 2;
+
+export type ValidatorRegistration = {
+  epoch: Epoch;
+  /** Preferred gas limit of validator */
+  gasLimit: number;
+};
+
+export class ValidatorRegistrationCache {
```

Comment:
might be able to combine this with `BeaconProposerCache` in the future but this requires apis to be consolidated first as proposed in https://github.com/ethereum/beacon-APIs/issues/435